lix/source-code/client/src/api.test.ts

```diff
@@ -1,205 +1,417 @@
+
+			const repoUrl = await findRepoRoot({
+				nodeishFs: fs,
+				path: "/src/routes/todo", // should find repo root from any path in the repo
+			})
+
+			expect(repoUrl).toBe("file:///")
+
+			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion -- test fails if repoUrl is null
+			const repository: Awaited<ReturnType<typeof openRepository>> = await openRepository(
+				repoUrl!,
+				{
+					nodeishFs: fs,
+					branch: "test-symlink",
+				}
+			)
 
-	it("can commit local modifications to the repo", async () => {
-		const statusPre = await repository.status({ filepath: "README.md" })
+			// test random repo action to make sure opening worked
+			const status = await repository.status("README.md")
 
-		expect(statusPre).toBe("*modified")
+			expect(status).toBe("unmodified")
+		})
 
-		await repository.add({ filepath: "README.md" })
-		await repository.commit({
-			author: { name: "tests", email: "test@inlang.dev" },
-			message: "test changes commit",
+		it("uses standard commit logic for non whitelisted repos", async () => {
+			const nonWhitelistedRepo = await openRepository(
+				"https://github.com/janfjohannes/unicode-bug-issues-1404",
+				{
+					nodeishFs: createNodeishMemoryFs(),
+				}
+			)
+			expect(nonWhitelistedRepo._experimentalFeatures).toStrictEqual({})
 		})
 
-		const statusPost = await repository.status({ filepath: "README.md" })
+		it("exposes proper origin", async () => {
+			const gitOrigin = await repository.getOrigin()
+			expect(gitOrigin).toBe("github.com/inlang/ci-test-repo.git")
+		})
 
-		expect(statusPost).toBe("unmodified")
-	})
+		it("finds initial commit of repo", async () => {
+			const fs = createNodeishMemoryFs()
 
-	it("can commit open repos without origin or git config (the case eg. on render.com or other deoployment scenarios)", async () => {
-		const fs = createNodeishMemoryFs()
+			const snapshot = JSON.parse(
+				readFileSync(resolve(__dirname, "../mocks/ci-test-repo-no-shallow.json"), {
+					encoding: "utf-8",
+				})
+			)
+			fromSnapshot(fs, snapshot)
 
-		const snapshot = JSON.parse(readFileSync("./mocks/ci-test-repo.json", { encoding: "utf-8" }))
-		fromSnapshot(fs, snapshot)
+			const firstHash = await repository.getFirstCommitHash()
+			expect(firstHash).toBe("244e3ce8c3335530ac0cd07e669b964bceb3b787")
+		})
 
-		await fs.rm("/.git/config")
+		it("exposes current branch", async () => {
+			const branch = await repository.getCurrentBranch()
+			expect(branch).toBe("main")
+		})
 
-		const repoUrl = await findRepoRoot({
-			nodeishFs: fs,
-			path: "/src/routes/todo", // should find repo root from any path in the repo
+		it("exposes remotes", async () => {
+			const remotes = await repository.listRemotes()
+			expect(remotes).toEqual([
+				{
+					remote: "origin",
+					url: "https://github.com/inlang/ci-test-repo",
+				},
+			])
 		})
 
-		expect(repoUrl).toBe("file:///")
+		it("exposes log", async () => {
+			// TODO: do we want lazy history?
```

Comment:
Shall we discuss this or put it into an issue?

Reply:
its just a reminder todo, to think about how history api should behave when touching this test, its currently not needed and does not need to be tracked as ticket